sdk/core/core-paging/src/getPagedAsyncIterator.ts

```diff
@@ -1,29 +1,9 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { PagedAsyncIterableIterator, PageSettings } from "@azure/core-paging";
+import { PagedAsyncIterableIterator, PageSettings, PagedResult } from "./models";
```

Comment:
Because this file is detected as a rename, I can't comment on non-changed lines of it, but I have some remarks on it, like :

1. the `metainfo` param feels unnecessary.
2. `getDefaultNextLink` being generic but then immediately casting to `any` is silly
3. Presuming non-paged results will have a `results` property seems like a bad assumption
4. There is a lot of `any` casts, which feels dangerous to do generically across all payloads

Reply:
I deleted and readded the file in git history so you can comment there now.

1- `metaInfo` is needed to determine whether we're processing a page with a collection or one with an object. This knowledge is important to decide how to yield the results, yielding an array is different from yielding a single object.
2- Good catch, fixed.
3- What do you mean by non-paged? all what this says is that the paged results is located in a `results` property in the response object which I feel makes sense. I am open for better default assumptions though. If you still disagree, I can make `processPage` required and not have assumptions there.
4- I removed all `any` casts except for the one to access `results`

Reply:
Update:
3- I removed the assumption since it does not show up in the REST guidelines
4- removing the assumption in 3 made all remaining `any`'s to be unnecessary, so no more `any`s.